### PR TITLE
 mpv: update to 0.33.0

### PR DESCRIPTION
**Pull request by concatime**

### CODE REVIEW

**Comment by ericonr** [comment]
This should be a single commit, but the list of changes should be kept in the commit message :)

Also, :c for removed sndio support

**Comment by concatime** [comment]
@ericonr understood

Also, we should wait for https://github.com/void-linux/void-packages/pull/26670 before merging this.

**Comment by ericonr** [comment]
I'm going to see if a revert of https://github.com/mpv-player/mpv/commit/71d218eae4b4d93ada34ff74906f71ad359c84bc builds, I really hate that they removed it :/

**Comment by ericonr** [comment]
Would you mind pulling the `libplacebo into this PR? For things that depend on each other, it makes sense to update multiple packages in a single PR

**Comment by concatime** [comment]
I will add OpenAL tomorrow.

**Comment by q66** [comment]
don't use the OpenAL backend in mpv, it has massive issues and you're better off just using pulse, or if you really want to keep sndio, probably `alsa-sndio` + `alsa` `ao` would work

**Comment by ericonr** [comment]
AFAIK `alsa-sndio` has multiple sync issues, and OpenAL was the upstream recommendation when people complained about sndio being removed.

**Comment by q66** [comment]
as far as i know, the `openal` backend is slow, experimental and severely broken with multichannel audio

there's a good reason it's disabled by the build system by default (even if openal is found on your system)

**Comment by concatime** [comment]
>  i don't see upstream ever recommending it anywhere

In commit removing OSS (https://github.com/mpv-player/mpv/commit/bca917f):
> Ancient Linux audio output. Apparently it survived until now, because
> some BSDs (but not all) had use of this. But these should work with
> ao_sdl **or ao_openal too** (that's why these AOs exist after all).

**Comment by concatime** [comment]
So, I added OpenAL as a build option, but not enabled by default.

**Comment by ericonr** [comment]
It might need atomic support with `libatomic`. Use `XBPS_TARGET_NO_ATOMIC8` as a conditional for `makedepends+=" libatomic-devel"`... At least I think that's the issue

**Comment by concatime** [comment]
I merged https://github.com/void-linux/void-packages/pull/26670 and https://github.com/void-linux/void-packages/pull/26678 into this request, with the right order.

**Review by ericonr — APPROVED** [review]
For what it's worth, this is working. `sndio` might be added back next release, but i don't think this update should wait on a way forward for sndio being decided so we could add a patch for it.

**Comment by concatime** [comment]
up

File: srcpkgs/mpv/template
Comment (ericonr) [inline srcpkgs/mpv/template:40]: Was removing Lua from default options intentional?

Comment (ericonr) [reply srcpkgs/mpv/template:40]: Sorry, viewing on phone and got confused!

**Comment by ericonr** [comment]
I'm waiting on feedback from others on patching sndio back, otherwise I can merge tomorrow morning.

Sorry for the delay.

**Comment by ericonr** [comment]
Thanks for working on this!

**Comment by stnby** [comment]
I happened to switch to sndio right after this commit. And got so confused by this https://docs.voidlinux.org/config/media/sndio.html#mpv I hope the support comes back as soon as possible. :)

**Pull request merged**
